### DISCUSSION TITLE
Changing ID pattern: How can I make Zettlr forget about the old one ?

### DISCUSSION THREAD

**Discussion by cmagam** [Zettlr/Zettlr#2900] [Q&A] [answered]
Hi folks,

First of all, many thanks for making Zettlr, it's a very nice piece of software.

I am currently fiddling with internal links using Zettelkasten IDs but I'm stuck with what is probably a trivial problem. 
I have opened a workspace with a bunch of old markdown files and I am trying to establish links between them. 
Problem: several of my files contains images that are named with the exact same pattern as the default Zettelkasten ID. As a result, the image file names (e.g. `![image](20180514173430.jpg)`) are interpreted as valid IDs by Zettlr. To make the matter worse, some images are used in several markdown files and generate duplicated IDs.

No sweat, I thought, let's change the Zettelkasten ID and regex patterns in the preferences. So I chose something a bit more unique that would not interfere with the names of my images. The problem is that Zettlr still seem to use the old (default) ID pattern when I try to link files with `[[]]` or when I do "copy ID". It looks like Zettlr is still using the old ID pattern for linking whereas the command to create a new ID (`CTl+L`) actually generates a new ID pattern. 

How can I make Zettlr forget about the old IDs when changing the ID pattern and make it rescan files for the new ones ? Is there a simple trick to do that ? 
I tried closing/opening workspaces, closing Zettlr with no luck.

I'm using Zettlr 2.0.3 both on Mac (Big Sur) and Windows 10.

Thanks for any help on what must probably be a simple thing.

have a good day

Chris

**Comment by nathanlesage** [comment, marked as answer]
The main problem is the FSAL cache which doesn't get removed after changing that. There's also an issue open about that (#2810) but until that's resolved, the simples way would be to …

0. Adapt the Zettelkasten ID as you would like to have it
1. Close Zettlr
2. Navigate to the app directory
3. Just remove the `fsal` directory in there

**Reply by cmagam** [reply]
Thanks a lot,

That did the trick :=) 
In fact I suspected something like this and I even tried to remove the `zettlr/Cache` folder but obviously it was not the right cache!
Anyway now that I started over with a clean ID pattern, I can move on 👍 
Big thanks,
cheers,
Chris
